tests/sga.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//==============================================================================
+// Imports
+//==============================================================================
+
+use ::demikernel::LibOS;
+use ::runtime::types::dmtr_sgarray_t;
+
+//==============================================================================
+// Constants
+//==============================================================================
+
+/// Size of scatter-gather arrays.
+const SGA_MAX_SIZE: usize = 1280;
```

Comment:
Might we want to test allocating different sizes, instead of always the same size?

Reply:
Nice! I've added tests for 64-byte SGA arrays, to cover the heap-allocated buffers in catnip. Also, as we've discussed, I'll add even more comprehensive tests after I'm done with the re-architecting of this memory management component.

Reply:
This is great, thanks!